scripts/build_renewable_profiles.py

```diff
@@ -180,220 +179,158 @@
   reached.
 
 """
-import logging
-from _helpers import configure_logging
-
-import os
-import atlite
-
-import numpy as np
-import xarray as xr
-import pandas as pd
-import multiprocessing as mp
-import matplotlib.pyplot as plt
 import progressbar as pgb
-
-from scipy.sparse import csr_matrix, vstack
+import geopandas as gpd
+import xarray as xr
+import numpy as np
+import atlite
+import logging
 from pypsa.geo import haversine
-from vresutils import landuse as vlanduse
-from vresutils.array import spdiag
-
-logger = logging.getLogger(__name__)
-
-bounds = dx = dy = config = paths = gebco = clc = natura = None
+from shapely.geometry import LineString
+import time
 
+from _helpers import configure_logging
 
-def init_globals(bounds_xXyY, n_dx, n_dy, n_config, n_paths):
-    # Late import so that the GDAL Context is only created in the new processes
-    global gl, gk, gdal
-    import glaes as gl
-    import geokit as gk
-    from osgeo import gdal as gdal
+logger = logging.getLogger(__name__)
 
-    # global in each process of the multiprocessing.Pool
-    global bounds, dx, dy, config, paths, gebco, clc, natura
 
-    bounds = gk.Extent.from_xXyY(bounds_xXyY)
-    dx = n_dx
-    dy = n_dy
-    config = n_config
-    paths = n_paths
+if __name__ == '__main__':
+    if 'snakemake' not in globals():
+        from _helpers import mock_snakemake
+        snakemake = mock_snakemake('build_renewable_profiles', technology='solar')
+    configure_logging(snakemake)
+    pgb.streams.wrap_stderr()
+    paths = snakemake.input
+    nprocesses = snakemake.config['atlite'].get('nprocesses')
+    noprogress = not snakemake.config['atlite'].get('show_progress', True)
+    config = snakemake.config['renewable'][snakemake.wildcards.technology]
+    resource = config['resource'] # pv panel config / wind turbine config
+    correction_factor = config.get('correction_factor', 1.)
+    capacity_per_sqkm = config['capacity_per_sqkm']
+    p_nom_max_meth = config.get('potential', 'conservative')
 
-    if "max_depth" in config:
-        gebco = gk.raster.loadRaster(paths["gebco"])
-        gebco.SetProjection(gk.srs.loadSRS(4326).ExportToWkt())
+    if isinstance(config.get("corine", {}), list):
+        config['corine'] = {'grid_codes': config['corine']}
 
-    clc = gk.raster.loadRaster(paths["corine"])
-    clc.SetProjection(gk.srs.loadSRS(3035).ExportToWkt())
+    if correction_factor != 1.:
+        logger.info(f'correction_factor is set as {correction_factor}')
 
-    natura = gk.raster.loadRaster(paths["natura"])
+    if not config.get('keep_all_available_areas', True):
+        logger.warning("Argument `keep_all_available_areas` is ignored. "
+                       "Continue with keeping all areas.")
```

Comment:
Is this `keep_all_available_areas` a new flag? Shouldn't we - if it is set to `False` -
rather `raise NotImplementedError("Argument 'keep_all_available_areas=False' is not implemented.")`

Reply:
It's an old flag, which is not available anymore. Raise an error would be a rough break, a warning that it's ignored from now on is more modest

Reply:
Hm, ok. I guess in general we should think about how to treat deprecated options (warn about them centrally, when to remove them, ...). If warnings for deprecated config options build up it will get messy over time.

Reply:
You are right, perhaps it's a better option to mention it in the release notes and not "during" the workflow?

Reply:
Yes! If the option is removed it has to be noted in the release notes.
If we want to notify the user of unsupported options, then we would think about a small scriptlet checks the `config.yaml` for consistency and deprecated options.

If you want to keep the warning, then I suggest you expand it:
```
if 'keep_all_available_areas' in config:
    # Warning
```

Reply:
Agree!

Reply:
Great. Cross-referencing the #158 issue which is open for a confic check script.